### ISSUE TITLE
Why is there a big difference between the two running times

### ISSUE THREAD

**Issue by PANFEI-CHENG** [rusty1s/pytorch_scatter#388]
I repeatedly execute the following code snippets in the debug console, why the running time is so different in the case of the same input.

######
start = torch.cuda.Event(enable_timing=True)
end = torch.cuda.Event(enable_timing=True)
start.record()
softmax_attn_flat = scatter_softmax(src=attn_flat, index=index_0, dim=0)
end.record()
torch.cuda.synchronize()
print(start.elapsed_time(end))
##9.083744049072266
start = torch.cuda.Event(enable_timing=True)
end = torch.cuda.Event(enable_timing=True)
start.record()
softmax_attn_flat = scatter_softmax(src=attn_flat, index=index_0, dim=0)
end.record()
torch.cuda.synchronize()
print(start.elapsed_time(end))
##178.6510467529297

**Comment by rusty1s** [comment]
I would personally run this in a loop and take the median/average. Also, you need to give the GPU some time to warm-up, so don't track the first few iterations.

**Comment by PANFEI-CHENG** [comment]
> I would personally run this in a loop and take the median/average. Also, you need to give the GPU some time to warm-up, so don't track the first few iterations.

thanks for your quick answer.I also suspected this at first. But I ran some other code ahead of time and nothing seemed to change.I have solved this problem by replacing the new version.
